mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -246,4 +242,127 @@ private static MlflowHostCredsProvider getHostCredsProviderFromTrackingUri(Strin
     }
     return provider;
   }
+
+  /**
+   * Uploads the given local file to the run's root artifact directory. For example,
+   *
+   *  logArtifact(runId, "/my/localModel")
+   *  listArtifacts(runId) // returns "localModel"
+   *
+   * @param runId Run ID of an existing MLflow run.
+   * @param localFile File to upload. Must exist, and must be a simple file (not a directory).
+   */
+  public void logArtifact(String runId, File localFile) {
+    getArtifactRepository(runId).logArtifact(localFile);
+  }
+
+  /**
+   * Uploads the given local file to an artifactPath within the run's root directory. For example,
+   *
+   *   logArtifact(runId, "/my/localModel", "model")
+   *   listArtifacts(runId, "model") // returns "model/localModel"
+   *
+   * (i.e., the localModel file is now available in model/localModel).
+   *
+   * @param runId Run ID of an existing MLflow run.
+   * @param localFile File to upload. Must exist, and must be a simple file (not a directory).
+   * @param artifactPath Artifact path relative to the run's root directory. Should NOT
+   *                     start with a /.
+   */
+  public void logArtifact(String runId, File localFile, String artifactPath) {
+    getArtifactRepository(runId).logArtifact(localFile, artifactPath);
+  }
+
+  /**
+   * Uploads all files within the given local director the run's root artifact directory.
+   * For example, if /my/local/dir/ contains two files "file1" and "file2", then
+   *
+   *  logArtifacts(runId, "/my/local/dir")
+   *  listArtifacts(runId) // returns "file1" and "file2"
```

Comment:
These may not format correctly in JavaDoc; you should check the generated doc to make sure it adds line breaks here, or use other formatting to make it work